packages/imperative/src/config/src/ProfileInfo.ts

```diff
@@ -201,6 +201,13 @@ export class ProfileInfo {
         }
 
         const mergedArgs = this.mergeArgsForProfile(desiredProfile, { getSecureVals: false });
+        if (options.forceUpdate && this.usingTeamConfig) {
+            const knownProperty = mergedArgs.knownArgs.find((v => v.argName === options.property));
+            const profPath = this.getTeamConfig().api.profiles.getProfilePathFromName(options.profileName);
+            if (!knownProperty.argLoc.jsonLoc.startsWith(profPath)) {
```

Comment:
If `options.property` is not a valid property in this profile layer, could this function error out on line 207?

I think optional chaining would handle that scenario:
```suggestion
            if (!knownProperty?.argLoc.jsonLoc.startsWith(profPath)) {
```